improver/calibration/rainforest_calibration.py

```diff
@@ -170,40 +179,56 @@ def __init__(self, model_config_dict: dict, threads: int = 1):
         Dictionary is of format::
 
             {
-                "-50.0" : {
-                    "lightgbm_model" : "<path_to_lightgbm_model_object>"
-                },
-                "-25.0" : {
-                    "lightgbm_model" : "<path_to_lightgbm_model_object>"
-                },
-                ...,
-                "50.0" : {
-                    "lightgbm_model" : "<path_to_lightgbm_model_object>"
+                "24": {
+                    "0.000010": {
+                        "lightgbm_model": "<path_to_lightgbm_model_object>",
+                        "treelite_model": "<path_to_treelite_model_object>"
+                    },
+                    "0.000050": {
+                        "lightgbm_model": "<path_to_lightgbm_model_object>",
+                        "treelite_model": "<path_to_treelite_model_object>"
+                    },
+                    "0.000100": {
+                        "lightgbm_model": "<path_to_lightgbm_model_object>",
+                        "treelite_model": "<path_to_treelite_model_object>"
+                    },
                 }
             }
 
-        The keys specify the error threshold value, while the associated values
-        are the path to the corresponding tree-model objects for that threshold.
+        The keys specify the lead times and model threshold values, while the
+        associated values are the path to the corresponding tree-model objects
+        for that lead time and threshold.
         """
         from lightgbm import Booster
 
-        # Dictionary keys represent error thresholds, however may be strings as they
-        # are sourced from json files. In order use these in processing, and to sort
-        # them in a sensible fashion, we shall cast the key values as float32.
-        sorted_model_config_dict = OrderedDict(
-            sorted({np.float32(k): v for k, v in model_config_dict.items()}.items())
-        )
+        # Model config is a nested dictionary. Keys of outer level are lead times, and
+        # keys of inner level are thresholds. Convert these to int and float.
+        sorted_model_config_dict = OrderedDict()
+        lead_time_keys = sorted([int(key) for key in model_config_dict.keys()])
+        for lead_time_key in lead_time_keys:
+            sorted_model_config_dict[lead_time_key] = OrderedDict()
+            lead_time_dict = model_config_dict[str(lead_time_key)]
+            sorted_model_config_dict[lead_time_key] = OrderedDict(
+                sorted({np.float32(k): v for k, v in lead_time_dict.items()}.items())
+            )
 
-        self.error_thresholds = np.array([*sorted_model_config_dict.keys()])
+        self.lead_times = np.array([*sorted_model_config_dict.keys()])
+        if len(self.lead_times) > 0:
+            self.model_thresholds = np.array(
+                [*sorted_model_config_dict[self.lead_times[0]].keys()]
+            )
+        else:
+            self.model_thresholds = np.array([])
```

Comment:
Although it would not happen in actual use, in some of the tests, the class is initialised by passing in an empty config, so we need to handle that case.

Reply:
Hmm... interesting. This is clearly an existing issue that has been made more apparent now that this has been generalised to allow lead-time variation.

Maybe a good middle ground here could be to raise a warning, that way we acknowledge this will fail, but still enable the efficient initialisation of the associated classes for testing the class functions that don't require model-objects.

Reply:
Thanks for adding in the warning. I'm happy that this has been addressed now.